ui/src/data/source.tsx

```diff
@@ -439,3 +571,226 @@ function processEntitiesResponse(
   }
   return nodes;
 }
+
+type EntityInstance = { [key: string]: tanagra.AttributeValue | null };
+
+function processEntityInstance(
+  attributeID: string,
+  instance: EntityInstance,
+  attributeHandler: (
+    key: string,
+    value: tanagra.AttributeValue | null
+  ) => boolean
+): DataEntry {
+  const data: DataEntry = {
+    key: 0,
+  };
+
+  for (const k in instance) {
+    const v = instance[k];
+    // TODO(tjennison): Add support for computed columns.
+    if (k === "standard_concept") {
+      data[k] = v ? "Standard" : "Source";
+    } else if (attributeHandler(k, v)) {
+      continue;
+    } else if (!v) {
+      data[k] = "";
+    } else if (isValid(v.int64Val)) {
+      data[k] = v.int64Val;
+    } else if (isValid(v.boolVal)) {
+      data[k] = v.boolVal;
+    } else if (isValid(v.stringVal)) {
+      data[k] = v.stringVal;
+    }
+  }
+
+  const key = data[attributeID];
+  if (!key || typeof key === "boolean") {
+    throw new Error(
+      `Key attribute "${attributeID}" not found in entity instance ${data}`
+    );
+  }
+  data.key = key;
+  return data;
+}
+
+// TODO(tjennison): Move this to BackendSource and make it private once the
+// count API uses have been converted.
+export function generateFilter(
+  source: Source,
+  filter: Filter | null,
+  fromPrimary: boolean
+): tanagra.Filter | null {
+  if (!filter) {
+    return null;
+  }
+
+  if (isArrayFilter(filter)) {
+    const operands = filter.operands
+      .map((o) => generateFilter(source, o, fromPrimary))
+      .filter(isValid);
+    if (operands.length === 0) {
+      return null;
+    }
+
+    let operator = tanagra.ArrayFilterOperator.Or;
```

Comment:
nit: consider extracting this so the value isn't dangling -- as an added bonus you can drop the else chain and just early return... will matter more as you add more operators potentially and the chain gets more complex.

Reply:
Done.